### ISSUE TITLE
Inconsistent errors for JSON-encoding NaNs with allow_nan=False

### ISSUE THREAD

**Issue by d874d704-1a46-47d2-84f3-2349b77eeff6** [python/cpython#86018]
BPO | [41852](https://bugs.python.org/issue41852)
--- | :---
Nosy | @serhiy-storchaka, @earwig

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2020-09-24.16:44:20.041>
labels = ['type-bug', '3.8', '3.9', '3.10', 'extension-modules', 'library']
title = 'Inconsistent errors for JSON-encoding NaNs with allow_nan=False'
updated_at = <Date 2020-09-24.17:05:07.621>
user = 'https://github.com/earwig'
```

bugs.python.org fields:
```python
activity = <Date 2020-09-24.17:05:07.621>
actor = 'serhiy.storchaka'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['Extension Modules', 'Library (Lib)']
creation = <Date 2020-09-24.16:44:20.041>
creator = 'earwig'
dependencies = []
files = []
hgrepos = []
issue_num = 41852
keywords = []
message_count = 1.0
messages = ['377450']
nosy_count = 2.0
nosy_names = ['serhiy.storchaka', 'earwig']
pr_nums = []
priority = 'normal'
resolution = None
stage = None
status = 'open'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue41852'
versions = ['Python 3.8', 'Python 3.9', 'Python 3.10']
```

</p></details>

* PR: gh-99172

**Comment by nineteendo** [comment]
This seems to be fixed:

```pytb
>>> json.dumps(float('nan'), allow_nan=False)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\wanne\AppData\Local\Programs\Python\Python312\Lib\json\__init__.py", line 238, in dumps
    **kw).encode(obj)
          ^^^^^^^^^^^
  File "C:\Users\wanne\AppData\Local\Programs\Python\Python312\Lib\json\encoder.py", line 200, in encode
    chunks = self.iterencode(o, _one_shot=True)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\wanne\AppData\Local\Programs\Python\Python312\Lib\json\encoder.py", line 258, in iterencode
    return _iterencode(o, 0)
           ^^^^^^^^^^^^^^^^^
ValueError: Out of range float values are not JSON compliant: nan
```

**Comment by picnixz** [comment]
It was reported in a separate issue #99925 and resolved by #99926. I think this one was just forgotten. I'll close it as completed, even though it's a duplicate (well technically, the other is a duplicate...)
